### PR TITLE
Merge Library into BookViewer app

### PR DESCRIPTION
**Pull request by agarg8487**
|PR checklist|Status|
|-----|:------:|
|Is issue created in GitHub?|[Yes](https://github.com/maqsoftware/GLEXP-Team-KitkitSchool-Hindi/issues/103)|
|Is code analysis task executed and issues resolved in the Java/Kotlin code?|Yes|
|Is code analysis task executed and issues resolved in the CPP/native code?|NA|
|Is the version code set in the AndroidManifest.xml or build.gradle file to 1 more than the version which has already been published?|No|
|Is code commenting done properly for the changes made?|Yes|
|Is correct variable naming convention followed?|Yes|
|Is auto formatting shortcut Ctrl+Alt+Shift+L executed in all files?|Yes|
|Are the message strings used in the application taken from strings.xml file?|No|
|Are the permissions which are not required removed from AndroidManifest.java file?|NA|
|Is the _uses-feature_ tag used for only must have feature in AndroidManifest.java file?|NA|
|Are the generated files correctly configured in .gitignore?|Yes|
|Are the release signing configurations removed from the gradle files?|NA|
|Is commented code removed from all files?|Yes|
|Are updated images compressed (< 100 KB)?|NA|
|Ensured there are no typos in any string. For example: Variable/method names, hardcoded strings, commit messages, PR titles, issues, code comments?|Yes|

### CODE REVIEW

File: bookviewer/proj.android-studio/app/build.gradle [hunk 1]
```diff
@@ -150,5 +150,17 @@ dependencies {
     compile fileTree(dir: 'libs', include: ['*.jar'])
     compile project(':libcocos2dx')
     compile project(':kitkitlogger')
-    //compile project(path: ':kitkitlogger', configuration: 'swahiliRelease')
-}
+//    Used by Library
+    compile 'com.android.support.constraint:constraint-layout:1.1.3'
+    androidTestCompile('com.android.support.test.espresso:espresso-core:2.2.2', {
+        exclude group: 'com.android.support', module: 'support-annotations'
+    })
```
Comment (ankitk20) [inline bookviewer/proj.android-studio/app/build.gradle]: Remove if not required

File: bookviewer/proj.android-studio/app/src/org/cocos2dx/cpp/BookApplication.java [hunk 2]
```diff
@@ -12,27 +18,44 @@
...
+        // method.
+        ImageLoaderConfiguration.Builder config = new ImageLoaderConfiguration.Builder(context);
+        config.threadPriority(Thread.NORM_PRIORITY - 2);
+        config.denyCacheImageMultipleSizesInMemory();
+        config.diskCacheFileNameGenerator(new Md5FileNameGenerator());
+        config.diskCacheSize(100 * 1024 * 1024); // 100 MiB
+        config.tasksProcessingOrder(QueueProcessingType.LIFO);
+
+        if (BuildConfig.DEBUG == true) {
+            config.writeDebugLogs(); // Remove for release app
```
Comment (ankitk20) [inline bookviewer/proj.android-studio/app/src/org/cocos2dx/cpp/BookApplication.java:34]: Remove == true

**Review by ankitk20 — CHANGES_REQUESTED** [review]

File: bookviewer/proj.android-studio/app/build.gradle (hunk 1 above)
Comment (agarg8487) [reply bookviewer/proj.android-studio/app/build.gradle]: Resolved.

File: bookviewer/proj.android-studio/app/src/org/cocos2dx/cpp/BookApplication.java (hunk 2 above)
Comment (agarg8487) [reply bookviewer/proj.android-studio/app/src/org/cocos2dx/cpp/BookApplication.java:34]: Resolved.

**Review by ankitk20 — APPROVED** [review]

**Review by harry6396 — APPROVED** [review]

**Review by mathewnik90 — APPROVED** [review]

File: bookviewer/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java [hunk 3]
```diff
@@ -77,33 +77,53 @@ public void run() {
...
+            editor.putString(key, val);
+            editor.apply();
+        } catch (Exception ignored) {
+
+        }
+    }
+
+    public void isStoragePermissionGranted() {
         if (Build.VERSION.SDK_INT >= 23) {
             if (checkSelfPermission(android.Manifest.permission.READ_EXTERNAL_STORAGE)
```
Comment (mathewnik90) [inline bookviewer/proj.android-studio/app/src/org/cocos2dx/cpp/AppActivity.java:119]: Update the if condition to be negative, to avoid empty if statement

**Pull request merged**